In [1]:
import os
import pandas as pd
print(os.getcwd())

c:\Users\lberm\OneDrive\Documents\Github\Gestalt_Question_Review\data


In [2]:
data_path = r"QuestionDataV2_06122025_classified.csv"
df =pd.read_csv(data_path)
print(df.columns)
df.head()

Index(['Unnamed: 0', 'Question Title', 'info.json', 'question.html',
       'server.js', 'solution.html', 'server.py', 'properties.js', 'question',
       'uuid', 'title', 'stem', 'topic', 'tags', 'prereqs', 'isAdaptive',
       'createdBy', 'qType', 'nSteps', 'updatedBy', 'difficulty', 'codelang',
       'resources', 'stepType', 'dificulty', 'relevant_courses', 'topics'],
      dtype='object')


Unnamed: 0 Question Title  \
0           0      3dMoment1   
1           1      3dMoment2   
2           2      3dMoment3   
3           3      3dMoment4   
4           4   Equilibrium1   

                                           info.json  \
0  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...   
1  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...   
2  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...   
3  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...   
4  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...   

                                       question.html  \
0  <pl-question-panel>\r\n    <pl-figure file-nam...   
1  <pl-question-panel>\r\n    <pl-figure file-nam...   
2  <pl-question-panel>\r\n    <pl-figure file-nam...   
3  <pl-question-panel>\r\n    <pl-figure file-nam...   
4  <pl-question-panel>\r\n<pl-figure file-name="3...   

                                           server.js  \
0  const math = require('mathjs');\r\n// const ma...   
1  const math = require('mathjs');\r\n// const ma...   
2  const math = require('mathjs');\r\n// const ma...   
3  const math = require('mathjs');\r\n// const ma...   
4  const math = require('mathjs');\r\n// const ma...   

                                       solution.html server.py properties.js  \
0  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
1  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
2  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
3  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
4  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   

                                            question  \
0  determine the moment about point o in ($n \cdo...   
1  Refer to the diagram labeled "3dMoment2.png". ...   
2  determine the resultant moment produced by the...   
3  determine the resultant moments produced by th...   
4  compute the missing force vector that will all...   

                                    uuid  ... qType nSteps updatedBy  \
0  8abcd91d6-64d1-4e89-b72d-ad2133f25b2f  ...   Num      1       NaN   
1  8abcd91d6-64d1-4e89-b72d-ad2133f25b2f  ...   Num      1       NaN   
2  8abcd91d6-64d1-4e89-b72d-ad2133f25b2f  ...   Num      1       NaN   
3  8abcd91d6-64d1-4e89-b72d-ad2133f25b2f  ...   Num      1       NaN   
4  8abcd91d6-64d1-4e89-b72d-ad2133f25b2f  ...   Num      1       NaN   

  difficulty    codelang  resources stepType dificulty  relevant_courses  \
0        1.0  Javascript        NaN      NaN       NaN    ME 010, ME 103   
1        1.0  Javascript        NaN      NaN       NaN    ME 103, ME 110   
2        1.0  Javascript        NaN      NaN       NaN               NaN   
3        1.0  Javascript        NaN      NaN       NaN    ME 010, ME 103   
4        1.0  Javascript        NaN      NaN       NaN            ME 010   

                                              topics  
0             Static Equilibrium, Distributed Forces  
1                            3D Rigid Body Mechanics  
2             Static Equilibrium, Distributed Forces  
3  Statics and Mechanics, Static Equilibrium, Vec...  
4                Static Equilibrium, Vector Analysis  

[5 rows x 27 columns]

In [4]:
columns_to_keep = ["Question Title","question.html","server.js","solution.html","server.py","properties.js","info.json","isAdaptive"]
df = df[columns_to_keep]
df.head()

Question Title                                      question.html  \
0      3dMoment1  <pl-question-panel>\r\n    <pl-figure file-nam...   
1      3dMoment2  <pl-question-panel>\r\n    <pl-figure file-nam...   
2      3dMoment3  <pl-question-panel>\r\n    <pl-figure file-nam...   
3      3dMoment4  <pl-question-panel>\r\n    <pl-figure file-nam...   
4   Equilibrium1  <pl-question-panel>\r\n<pl-figure file-name="3...   

                                           server.js  \
0  const math = require('mathjs');\r\n// const ma...   
1  const math = require('mathjs');\r\n// const ma...   
2  const math = require('mathjs');\r\n// const ma...   
3  const math = require('mathjs');\r\n// const ma...   
4  const math = require('mathjs');\r\n// const ma...   

                                       solution.html server.py properties.js  \
0  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
1  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
2  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
3  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   
4  <pl-solution-panel>\r\n    <pl-figure file-nam...       NaN           NaN   

                                           info.json  isAdaptive  
0  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...        True  
1  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...        True  
2  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...        True  
3  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...        True  
4  {\r\n    "uuid": "8abcd91d6-64d1-4e89-b72d-ad2...        True

In [5]:
row_num,col_num = df.shape
adaptive_questions_count = df["isAdaptive"].sum()
print(f"""The number of adaptive questions is {adaptive_questions_count} \n while non adaptive questions are {row_num-adaptive_questions_count}
"""
)
py_filtered = df[(df["isAdaptive"]) & (df["server.py"].notna())]
js_filtered = df[(df["isAdaptive"]) & (df["server.js"].notna())]

The number of adaptive questions is 158 
 while non adaptive questions are 105



In [14]:
import re
pattern = r'</pl-[a-z\-]+>'
unique_tags = set()
filtered = df[~df["isAdaptive"].isna()]
for text in filtered["question.html"]:
    if isinstance(text, str):
        result = re.findall(pattern, text)
        # if "</pl-mca-adaptive>" in result:
        #     print(text)
        unique_tags.update(result)
        
print(unique_tags)

{'</pl-figure>', '</pl-mcq-adaptive>', '</pl-matrix-latex>', '</pl-matrix-input>', '</pl-number-input>', '</pl-question-panel>', '</pl-mca-adaptive>', '</pl-answer>', '</pl-quest>', '</pl-static-text>', '</pl-number-input-fixed>', '</pl-multiple-choice>', '</pl-prompt>', '</pl-checkbox>'}


## Notes
- info.json: Originally held question metadata, now empty. Not essential for generation but should be repopulated.
- question.html: HTML template that renders the student-facing question (figures, inputs, choices, etc.).
- server files: Backend logic written in either JavaScript or Python.
-Total questions: 320 (used for semantic search); not guaranteed to reflect the current total.
- Adaptive questions: 158 — dynamically generated (e.g. numerical computations).
- Non-adaptive questions: 162 — static (e.g. multiple choice).
- Code files: 7 Python files, 139 JavaScript files related to questions.